sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -830,11 +838,60 @@ public void populateDisplayData(DisplayData.Builder builder) {
       inner.populateDisplayData(builder);
     }
 
+    private boolean hasStatementAndSetter() {
+      return inner.getStatement() != null && inner.getPreparedStatementSetter() != null;
+    }
+
     @Override
     public PDone expand(PCollection<T> input) {
+
+      if (input.hasSchema() && !hasStatementAndSetter()) {
+        Schema schema = input.getSchema();
+
+        inner = inner.withStatement(generateStatement(inner.getTable(), schema.getFields()));
+        inner = inner.withPreparedStatementSetter(((element, preparedStatement) -> {
+          // assuming we have PCollection of <Row>. We don't have Row then we will use toRowFunction().
+          // TODO: use toRowFunction() to convert other data types into Row
+          mapAccordingToSchema((Row) element, preparedStatement, schema); // Type casting element into Row is temporary untill we implement usage of toRowFunction()
+        }));
+      }
+
```

Comment:
if there is schema and statement is not present then we generate our own statement and setter method.

Type casting element into Row is temporary until we implement usage of `toRowFunction().`